### PR TITLE
Make gtc:numpy work with all OIR optimizations

### LINKED ISSUE #504: gtc:numpy backend generates incorrect code after after some oir passes

### PR DESCRIPTION
**Pull request by jdahm**
## Description

Fixes an error in `local_var_inside_nested_conditional` and edit the `gtc:numpy` backend to make `VectorTemp`s from scalarized temporaries.

With this the `gtc:numpy` backend passes all tests.

Resolves #504.

### CODE REVIEW

File: tests/test_integration/stencil_definitions.py [hunk 1]
```diff
@@ -16,14 +16,32 @@
...
 import numpy as np
 
-import gt4py as gt
-from gt4py import analysis as gt_analysis
-from gt4py import backend as gt_backend
 from gt4py import gtscript
-from gt4py import ir as gt_ir
-from gt4py import storage as gt_storage
 from gt4py import utils as gt_utils
-from gt4py.definitions import Extent, StencilID
```
Comment (jdahm) [inline tests/test_integration/stencil_definitions.py:26]: Most of these were unused...

File: tests/test_integration/stencil_definitions.py [hunk 2]
```diff
@@ -307,7 +325,7 @@ def local_var_inside_nested_conditional(in_storage: Field3D, out_storage: Field3
     with computation(FORWARD), interval(2, None):
         if in_storage[0, 0, 0] < 0:
             local_var = 6
-            out_storage[0, 0, 0] = local_var + mid_storage
+            out_storage[0, 0, 0] = local_var
```
Comment (jdahm) [inline tests/test_integration/stencil_definitions.py:328]: This test actually had an error to begin with, but was never caught before. This fixes it.

File: src/gtc/python/npir_gen.py [hunk 3]
```diff
@@ -230,18 +230,21 @@ def visit_EmptyTemp(
...
             mask = self.visit(node.mask)
-            mask_def = f"{mask_name}_ = np.full((I - i, J - j, K - k), {mask})\n"
+            ksize = "1" if is_serial else "K - k"
+            mask_def = f"{mask_name}_ = np.full((I - i, J - j, {ksize}), {mask})\n"
         else:
             mask_name = node.mask_name
             mask = self.visit(node.mask)
             mask_def = f"{mask_name}_ = {mask}\n"
-        return self.generic_visit(node, mask_def=mask_def, **kwargs)
+        return self.generic_visit(node, mask_def=mask_def, is_serial=is_serial, **kwargs)
```
Comment (DropD) [inline src/gtc/python/npir_gen.py:247]: If this is the only use of `is_serial` then why not go back to just passing it through via `kwargs`?

File: src/gtc/python/npir_gen.py [hunk 4]
```diff
@@ -230,18 +230,21 @@ def visit_EmptyTemp(
...
+        self, node: npir.MaskBlock, *, is_serial: bool = False, **kwargs
+    ) -> Union[str, Collection[str]]:
         if isinstance(node.mask, npir.FieldSlice):
             mask_def = ""
         elif isinstance(node.mask, npir.BroadCast):
             mask_name = node.mask_name
             mask = self.visit(node.mask)
-            mask_def = f"{mask_name}_ = np.full((I - i, J - j, K - k), {mask})\n"
+            ksize = "1" if is_serial else "K - k"
+            mask_def = f"{mask_name}_ = np.full((I - i, J - j, {ksize}), {mask})\n"
```
Comment (DropD) [inline src/gtc/python/npir_gen.py:242]:
```suggestion
            k_size = "1" if is_serial else "K - k"
            mask_def = f"{mask_name}_ = np.full((I - i, J - j, {k_size}), {mask})\n"
```

just for consistency / searchability.

**Review by DropD — COMMENTED** [review]
Just a few small things, looks good overall.

**Pull request merged**